src/main/java/com/onarandombox/MultiverseCore/destination/DestinationsProvider.java

```diff
@@ -45,13 +42,7 @@ public DestinationsProvider(@NotNull MultiverseCore plugin) {
      */
     public void registerDestination(@NotNull Destination<?> destination) {
         this.destinationMap.put(destination.getIdentifier(), destination);
-        this.registerDestinationPerms(destination);
-    }
-
-    private void registerDestinationPerms(@NotNull Destination<?> destination) {
-        PluginManager pluginManager = this.plugin.getServer().getPluginManager();
-        pluginManager.addPermission(new Permission(PERMISSION_PREFIX + "self." + destination.getIdentifier()));
-        pluginManager.addPermission(new Permission(PERMISSION_PREFIX + "other." + destination.getIdentifier()));
+        this.plugin.getPermissionsTool().registerDestinationTeleportPermissions(destination);
```

Comment:
This is great. Creating permissions should not be the concern of DestinationsProvider.